src/controllers/notifications.js

```diff
@@ -70,7 +71,19 @@ const model = require('../models/notifications')
 exports.post = function * postNotification () {
   const notification = yield requestUtil.validateBody(this, 'Notification')
   log.debug('Got notification: ' + JSON.stringify(notification))
-  yield model.processNotification(notification)
-
+  try {
+    yield model.processNotification(notification)
+  } catch (e) {
+    // If this receives UnacceptableExpiryError, do not respond with HTML error 422.
+    // Doing so would cause Ledger to reply with the same message, creating an infinite loop.
+    if (e instanceof UnacceptableExpiryError) {
```

Comment:
@lumberj I agree.  The model shouldn't be throwing an exception for situations like this.  It should be using a different way of error handling mechanism.  The problem is that it relies on throwing an error currently, and fixing that would be a refactoring job that's out of this PR's scope.